Use real database

ilopmar opened the pull request:
The configuration explained in this PR won't be valid until https://github.com/micronaut-projects/micronaut-core/pull/469 is merged.
Once it's merged make a manual test to make sure that the default configuration for the database is working.

ilopmar committed:
Use Micronaut build-snapshot version

ilopmar committed:
Explain how to use Postgresl

sdelamo reviewed (changes requested):
dataSourceExternal snippte should be directly in application.yml and the guide should be updated accordingly.

ilopmar committed:
Define the externalized configuration in application.yml

ilopmar committed:
Use back ticks to escape the connection url

ilopmar commented:
I've added a new commit to escape the database url connection with back ticks as this is the final character we're going to use. See https://github.com/micronaut-projects/micronaut-core/commit/4965ae7
I've also rebased the changes in master.
